src/derivatives/elementwise.jl

```diff
@@ -311,12 +383,25 @@ end
         end
     else
         a, b = input
+        p = 0
         if size(a) == size(b)
-            istracked(a) && diffresult_increment_deriv!(a, output_deriv, results, 1)
-            istracked(b) && diffresult_increment_deriv!(b, output_deriv, results, 2)
+            if istracked(a)
+                p += 1
+                diffresult_increment_deriv!(a, output_deriv, results, p)
+            end
+            if istracked(b)
+                p += 1
+                diffresult_increment_deriv!(b, output_deriv, results, p)
+            end
         else
-            istracked(a) && diffresult_increment_deriv!(a, output_deriv, results, 1, a_bound)
-            istracked(b) && diffresult_increment_deriv!(b, output_deriv, results, 2, b_bound)
+            if istracked(a)
+                p += 1
+                diffresult_increment_deriv!(a, output_deriv, results, p, a_bound)
```

Comment:
To extract the correct partial. If `a` is tracked, its corresponding partial has index `p = 1` but if only `b` is tracked, the first partial (`p = 1`) corresponds to `b`. And if both `a` and `b` are tracked, `p = 1` corresponds to `a` and `p = 2` to `b`. So incrementing `p` in the branches allows us to avoid checking and handling all three scenarios separately.

Note that on the master branch `p = 1` for `a` and `p = 2` for `b` are hardcoded. That only works because on the master branch *always* the partials wrt to both arguments are computed and stored, even if only one argument is tracked.